### PR TITLE
Adding building container capabilities

### PR DESCRIPTION
**Pull request by unlucio**

### CODE REVIEW

File: src/docker.js [hunk 1]
```diff
@@ -21,41 +63,47 @@ if (fs.existsSync('/tmp/docker.sock')) {
  */
 docker.buildImage = function(project, tarPath, imageId, buildId, buildLogger, dockerOptions) {
   var deferred = Q.defer();
+  var buildTag = imageId + ((dockerOptions.dockerfile) ? '-builder' : '');
```
Comment (odino) [inline src/docker.js]: tag

File: src/docker.js [hunk 2]
```diff
@@ -21,41 +63,47 @@ if (fs.existsSync('/tmp/docker.sock')) {
...
-        if (result.error) {
-          buildLogger.error('[%s] %s', buildId, result.error)
-          deferred.reject(result.error);
-          return;
-        }
+      try {
+        result = JSON.parse(out.toString('utf-8'));
+      } catch (err) {
+        buildLogger.error('[%s] %s', buildTag, err);
+      }
```
Comment (odino) [inline src/docker.js]: why removing the rejection?

**Comment by odino** [comment]
So the config could look like:

``` yaml
project:
  build:
    dockerfile: dockerfile.build
    extract: /src
```

We should also update the README.

File: src/docker.js [hunk 3]
```diff
@@ -14,6 +13,49 @@ if (fs.existsSync('/tmp/docker.sock')) {
...
+      delete dockerOptions.dockerfile;
+      var name = builderId.replace('/', '_').replace(':', '__') + '-retrive';
+
+      docker.client.createContainer({Image: builderId, name: name, Cmd: ['/bin/sh']}, function (err, container) {
+        if (err) {
+          reject(err);
+          return;
+        }
+
+        container.start(function (error) {
```
Comment (odino) [inline src/docker.js]: do we really need to start the container?

File: src/docker.js [hunk 4]
```diff
@@ -14,6 +13,49 @@ if (fs.existsSync('/tmp/docker.sock')) {
   docker.client = new Docker({host: require('netroute').getGateway(), port: 2375});
 }
 
+function extractAndRepackage(project, imageId, builderId, buildId, buildLogger, dockerOptions) {
+  return function() {
+    return Q.Promise(function(resolve, reject) {
+      delete dockerOptions.dockerfile;
+      var name = builderId.replace('/', '_').replace(':', '__') + '-retrive';
```
Comment (odino) [inline src/docker.js]: this could be 'uuid + builder'

**Comment by unlucio** [comment]
@odino ok let's figure out how to test it.

**Comment by odino** [comment]
lets run it on one of our private secondary containers?

**Pull request merged**